cassandra-dac/src/test/java/org/sunbird/cassandra/CassandraOperationImplTest.java

```diff
@@ -0,0 +1,635 @@
+                ResponseCode.invalidPropertyError.getErrorCode(),
+                JsonKey.UNKNOWN_IDENTIFIER,
+                ResponseCode.CLIENT_ERROR.getResponseCode()));
+
+    Throwable exception = null;
+    try {
+      operation.updateRecord(cassandraKeySpace, "address", dummyAddress);
+    } catch (Exception exp) {
+      exception = exp;
+    }
+    Object[] args = {""};
+    assertEquals(
+        new MessageFormat(ResponseCode.invalidPropertyError.getErrorMessage()).format(args),
+        exception.getMessage());
+  }
+
+  @Test
+  public void testGetAllRecordsSuccess() throws Exception {
+
+    List<Row> rows = new ArrayList<>();
+    Row row = Mockito.mock(Row.class);
+    rows.add(row);
+    when(resultSet.all()).thenReturn(rows);
+
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenReturn(boundStatement);
+
+    Response response = operation.getAllRecords(cassandraKeySpace, "address");
+    assertTrue(response.getResult().size() > 0);
+  }
+
+  @Test
+  public void testGetAllRecordsFailure() throws Exception {
+
+    when(session.execute(selectQuery))
+        .thenThrow(
+            new ProjectCommonException(
+                ResponseCode.SERVER_ERROR.getErrorCode(),
+                ResponseCode.SERVER_ERROR.getErrorMessage(),
+                ResponseCode.SERVER_ERROR.getResponseCode()));
+
+    List<Row> rows = new ArrayList<>();
+    Row row = Mockito.mock(Row.class);
+    rows.add(row);
+    when(resultSet.all()).thenReturn(rows);
+
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenReturn(boundStatement);
+
+    Throwable exception = null;
+    try {
+      operation.getAllRecords(cassandraKeySpace, "address");
+    } catch (Exception ex) {
+      exception = ex;
+    }
+    assertTrue(
+        (((ProjectCommonException) exception).getResponseCode())
+            == ResponseCode.SERVER_ERROR.getResponseCode());
+  }
+
+  @Test
+  public void testGetPropertiesValueSuccessById() throws Exception {
+
+    List<Row> rows = new ArrayList<>();
+    Row row = Mockito.mock(Row.class);
+    rows.add(row);
+    when(resultSet.all()).thenReturn(rows);
+
+    when(session.execute(boundStatement.bind("123"))).thenReturn(resultSet);
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenReturn(boundStatement);
+    Response response =
+        operation.getPropertiesValueById(
+            cassandraKeySpace, "address", "123", JsonKey.ID, JsonKey.CITY, JsonKey.ADD_TYPE);
+    assertTrue(response.getResult().size() > 0);
+  }
+
+  @Test
+  public void testGetPropertiesValueFailureById() throws Exception {
+
+    Throwable exception = null;
+    PowerMockito.whenNew(BoundStatement.class)
+        .withArguments(Mockito.any(PreparedStatement.class))
+        .thenThrow(
+            new ProjectCommonException(
```

Comment:
ProjectCommonException.throwServerErrorException - can use - equivalent method throughout the class